### PR TITLE
ci: simplify gae flexible requirements

### PR DESCRIPTION
**Pull request by kweinmeister**
There are now directories for [flexible](https://github.com/GoogleCloudPlatform/python-docs-samples/tree/main/appengine/flexible) (3.8+) and [flexible_python37_and_earlier](https://github.com/GoogleCloudPlatform/python-docs-samples/tree/main/appengine/flexible_python37_and_earlier) (<=3.7).

Knowing this, we can simplify our builds and dependencies. This PR:
- Skips running the 3.8+ samples (flexible) on 3.7 builds by updating `noxfile_config.py`.
- Removes unnecessary conditional logic in `requirements.txt`

### CODE REVIEW

File: appengine/flexible/analytics/noxfile_config.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Google LLC
```
Comment (engelke) [inline appengine/flexible/analytics/noxfile_config.py:1]: It seems to me that all the noxfile_config.py files under appengine/flexible can be replaced by a single one at appengine/flexible/noxfile_config.py. They appear the same from my scan.

**Review by engelke — COMMENTED** [review]
And how about a noxfile_config.py under appengine/flexible_python37_and_earlier that excludes all Python version > 3.7, since they are in appengine/flexible. All that should really simplify those conditional requirements.

**Review by engelke — APPROVED** [review]
I'm suggesting changes, but I'm fine without them too. It's up to you.

LGTM

**Review by kennethye1 — APPROVED** [review]
LGTM

**Comment by kweinmeister** [comment]
Thanks @engelke - will hold on merging this until I make the changes you suggested.

**Pull request merged**